### PR TITLE
[SPARK-50619][SQL] Refactor VariantGet.cast to pack the cast arguments

### PR DESCRIPTION
**Pull request by chenhao-db**
### What changes were proposed in this pull request?

As the title. It refactors the code for simplification.

### Why are the changes needed?

The refactor will make it simpler for the shredded user to use `VariantGet.cast`.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Existing tests.

### Was this patch authored or co-authored using generative AI tooling?

No.

### CODE REVIEW

**Comment by chenhao-db** [comment]
@cashmand @cloud-fan could you help review? Thanks!

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/variant/variantExpressions.scala [hunk 1]
```diff
@@ -323,6 +321,12 @@ case class VariantGet(
   override def withTimeZone(timeZoneId: String): VariantGet = copy(timeZoneId = Option(timeZoneId))
 }
 
+// Several parameters used by `VariantGet.cast`. Packed together to simplify parameter passing.
+case class VariantCastArgs(
+    failOnError: Boolean,
+    zoneStr: Option[String],
```
Comment (cloud-fan) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/variant/variantExpressions.scala:327]: why do we need the original timezone string if we already have the ZoneId?

Comment (cloud-fan) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/variant/variantExpressions.scala:327]: oh it's to create the `Cast` expression

**Review by cloud-fan — APPROVED** [review]

**Comment by chenhao-db** [comment]
@cloud-fan could you help merge it? Thanks!

**Comment by cloud-fan** [comment]
thanks, merging to master!

**Pull request closed without merging**